service/worker/archiver/client.go

```diff
@@ -103,11 +162,12 @@ func (c *client) Archive(request *ArchiveRequest) error {
 		DecisionTaskStartToCloseTimeout: workflowTaskStartToCloseTimeout,
 		WorkflowIDReusePolicy:           cclient.WorkflowIDReusePolicyAllowDuplicate,
 	}
-	_, err := c.cadenceClient.SignalWithStartWorkflow(context.Background(), workflowID, signalName, *request, workflowOptions, archivalWorkflowFnName, nil)
+	_, err := c.cadenceClient.SignalWithStartWorkflow(ctx, workflowID, signalName, *request, workflowOptions, archivalWorkflowFnName, nil)
```

Comment:
I think passing ctx here will not work because if inline failed ctx will be timed out at this point

Reply:
Changed to context.Background() for now. Question is after we have a reliable estimate of time needed by typical inline archival, do we want to change back to `ctx`. Basically if inline already takes too much time and blocking other tasks, do we want to spend more time to send the signal or just let it picked up by DB scanner.